internal/builtinfunctions/functions_test.go

```diff
@@ -0,0 +1,471 @@
+		nil,
+	},
+	"string-to-float-positive": {
+		"stringToFloat",
+		[]any{"11.1"},
+		false,
+		11.1,
+	},
+	"string-to-float-negative": {
+		"stringToFloat",
+		[]any{"-11.1"},
+		false,
+		-11.1,
+	},
+	"string-to-float-zero": {
+		"stringToFloat",
+		[]any{"0"},
+		false,
+		0.0,
+	},
+	"string-to-float-scientific-notation": {
+		"stringToFloat",
+		[]any{"5e+5"},
+		false,
+		500000.0,
+	},
+	"string-to-float-error-1": {
+		"stringToFloat",
+		[]any{""},
+		true,
+		nil,
+	},
+	"string-to-float-error-2": {
+		"stringToFloat",
+		[]any{"ten"},
+		true,
+		nil,
+	},
+	"string-to-bool-1": {
+		"stringToBool",
+		[]any{"true"},
+		false,
+		true,
+	},
+	"string-to-bool-2": {
+		"stringToBool",
+		[]any{"True"},
+		false,
+		true,
+	},
+	"string-to-bool-3": {
+		"stringToBool",
+		[]any{"TRUE"},
+		false,
+		true,
+	},
+	"string-to-bool-4": {
+		"stringToBool",
+		[]any{"t"},
+		false,
+		true,
+	},
+	"string-to-bool-5": {
+		"stringToBool",
+		[]any{"T"},
+		false,
+		true,
+	},
+	"string-to-bool-6": {
+		"stringToBool",
+		[]any{"1"},
+		false,
+		true,
+	},
+	"string-to-bool-7": {
+		"stringToBool",
+		[]any{"false"},
+		false,
+		false,
+	},
+	"string-to-bool-8": {
+		"stringToBool",
+		[]any{"False"},
+		false,
+		false,
+	},
+	"string-to-bool-9": {
+		"stringToBool",
+		[]any{"FALSE"},
+		false,
+		false,
+	},
+	"string-to-bool-10": {
+		"stringToBool",
+		[]any{"f"},
+		false,
+		false,
+	},
+	"string-to-bool-11": {
+		"stringToBool",
+		[]any{"F"},
+		false,
+		false,
+	},
+	"string-to-bool-12": {
+		"stringToBool",
+		[]any{"0"},
+		false,
+		false,
+	},
+	"string-to-bool-error-1": {
+		"stringToBool",
+		[]any{""},
+		true,
+		nil,
+	},
+	"string-to-bool-error-2": {
+		"stringToBool",
+		[]any{"abc"},
+		true,
+		nil,
+	},
+	"ceil-1": {
+		"ceil",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"ceil-2": {
+		"ceil",
+		[]any{0.1},
+		false,
+		1.0,
+	},
+	"ceil-3": {
+		"ceil",
+		[]any{-0.1},
+		false,
+		0.0,
+	},
+	"floor-1": {
+		"floor",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"floor-2": {
+		"floor",
+		[]any{1.1},
+		false,
+		1.0,
+	},
+	"floor-3": {
+		"floor",
+		[]any{2.9999},
+		false,
+		2.0,
+	},
+	"floor-4": {
+		"floor",
+		[]any{-0.1},
+		false,
+		-1.0,
+	},
+	"round-1": {
+		"round",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"round-2": {
+		"round",
+		[]any{0.1},
+		false,
+		0.0,
+	},
+	"round-3": {
+		"round",
+		[]any{0.4},
+		false,
+		0.0,
+	},
+	"round-4": {
+		"round",
+		[]any{0.5},
+		false,
+		1.0,
+	},
+	"round-5": {
+		"round",
+		[]any{0.9},
+		false,
+		1.0,
+	},
+	"round-6": {
+		"round",
+		[]any{-0.9},
+		false,
+		-1.0,
+	},
+	"round-7": {
+		"round",
+		[]any{-0.5},
+		false,
+		-1.0,
+	},
+	"round-8": {
+		"round",
+		[]any{-0.01},
+		false,
+		0.0,
+	},
+	"abs-1": {
+		"abs",
+		[]any{0.0},
+		false,
+		0.0,
+	},
+	"abs-2": {
+		"abs",
+		[]any{-1.0},
+		false,
+		1.0,
+	},
+	"abs-3": {
+		"abs",
+		[]any{1.0},
+		false,
+		1.0,
+	},
+	"to-lower-1": {
+		"toLower",
+		[]any{""},
+		false,
+		"",
+	},
```

Comment:
Mixed-case trials for upper/lower would probably be good, rather than just blank/already-{upper|lower}/opposite-case.